src/core_plugins/input_control_vis/public/components/vis/list_control.js

```diff
@@ -77,25 +80,32 @@ class ListControlUi extends Component {
       );
     }
 
-    const options = this.props.options.map(option => {
-      return {
-        label: option.label,
-        value: option.value,
-        ['data-test-subj']: `option_${option.value.replace(' ', '_')}`
-      };
-    });
-
     return (
       <EuiComboBox
         placeholder={intl.formatMessage({
           id: 'inputControl.vis.listControl.selectPlaceholder',
           defaultMessage: 'Select...'
         })}
-        options={options}
+        options={this.props.options.map(option => {
+          return {
+            label: this.props.formatOptionLabel(option).toString(),
+            value: option,
+            ['data-test-subj']: `option_${option.toString().replace(' ', '_')}`
+          };
+        })
+          .sort((a, b) => {
+            return a.label.toLowerCase().localeCompare(b.label.toLowerCase());
+          })
+        }
```

Comment:
nit: Why you have moved this map in the JSX? Don't you think it loose a bit of readability if written here?